src/routes/transactions.rs

```diff
@@ -1,10 +1,14 @@
+use crate::cache_resp;
+use crate::cache::ServiceCache;
 use crate::services::transactions;
 use rocket::response::content;
 use anyhow::Result;
 
 #[get("/transactions/<safe_address>")]
-pub fn all(safe_address: String) -> Result<content::Json<String>> {
-    Ok(content::Json(transactions::get_all_transactions(safe_address)?))
+pub fn all(cache: ServiceCache, safe_address: String) -> Result<content::Json<String>> {
+    cache.cache_resp(&safe_address, 60 * 2, || { 
```

Comment:
Maybe it would be good to parametrise the cache timeouts to have a single place of control for these values.